### PR TITLE
Remove 'TemplateRef' abstraction

### PR DESCRIPTION
**Pull request by nmattia**
This removes the `TemplateRef` abstraction which turned out to be really inconvenient to use. Since introducing `TemplateRef` I also realized that `Ref<T>` actually carries phantom type data, making most of the `TemplateRef` benefits moot.

### CODE REVIEW

File: src/frontend/src/components/anchorInput.ts [hunk 1]
```diff
@@ -37,6 +37,10 @@ export const mkAnchorInput = (
         @contextmenu=${inputFilter(isDigits, onBadInput)}
         @drop=${inputFilter(isDigits, onBadInput)}
         @focusout=${inputFilter(isDigits, onBadInput)}
+        @keypress=${onKeyPress ??
```
Comment (peterpeterparker) [inline src/frontend/src/components/anchorInput.ts:40]: doesn't lit don't bind the keypress if undefined is provided - i.e. a fallback is really needed?

File: src/frontend/src/flows/login/unknownAnchor.ts [hunk 2]
```diff
@@ -100,39 +109,41 @@ const initRecovery = (connection: Connection) => {
...
 
-  userNumberInput.onkeypress = (e) => {
-    // submit if user hits enter
-    if (e.key === "Enter") {
-      e.preventDefault();
-      loginButton.click();
-    }
-  };
+  withRef(userNumberInput, (userNumberInput) => {
+    userNumberInput.onkeypress = (e) => {
```
Comment (peterpeterparker) [inline src/frontend/src/flows/login/unknownAnchor.ts:120]: isn't this code duplicated? worth extracting a function?

File: src/frontend/src/utils/utils.ts [hunk 3]
```diff
@@ -1,3 +1,20 @@
+import { Ref } from "lit-html/directives/ref.js";
```
Comment (peterpeterparker) [inline src/frontend/src/utils/utils.ts:1]: not sure about this import, you might have issue someday if you switch to a module based bundler - just a feeling though

File: src/frontend/src/flows/login/unknownAnchor.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/login/unknownAnchor.ts:120]: yes, will do in a separate PR

File: src/frontend/src/utils/utils.ts [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+
+// Read a "lit-html" ref, showing an error message (in the console) in case the
+// element is not available.
+export function withRef<A, B>(ref: Ref<A>, f: (val: A) => B): B | undefined {
+  const value = ref.value;
+
+  if (value === undefined) {
```
[the remark this answers is not in the document]
Comment (nmattia) [reply src/frontend/src/utils/utils.ts:8]: no, the `value` can only be `undefined` (I think `null` would show up in the types)

File: src/frontend/src/components/anchorInput.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/components/anchorInput.ts:40]: good question, will check

**Review by peterpeterparker — APPROVED** [review]
LGTM

File: src/frontend/src/flows/login/unknownAnchor.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/login/unknownAnchor.ts:120]: you're right! updated

**Pull request merged**